docs/architecture/execution-engines.md

```diff
@@ -0,0 +1,42 @@
+## Execution Engines(?)
+
+A workload management system for the scheduling and running of jobs.  The typical enterprise requires support for multiple operating systems.  A core tenet of the project is to leverage both open source projects and commercial cloud products and their capabilities where we can. 
+TODO: Tie back to targeted use cases and typical enterprise
+
+Tier 1 supported environment is Linux and a typical use case of web serving and associated backend systems.  
+
+Tier 2 supported environment is Mac OSX for iOS and desktop clients.  We may also support Windows in the future.
+
+For tier 2 supported environments, we use Jenkins as our execution engine to handle scheduling across Mac OSX slaves.  Jenkins support across many OSes is useful here.
```

Comment:
mostly because I didn't want to talk about Jenkins and focus on the other engines.  This was for the purpose of a review.  I should actually expand the Jenkins part and do this in order.